connector/connect/server/src/main/scala/org/apache/spark/sql/connect/planner/SparkConnectPlanner.scala

```diff
@@ -616,6 +618,38 @@ class SparkConnectPlanner(val session: SparkSession) {
       .logicalPlan
   }
 
+  private def transformFlatMapGroupsWithState(rel: proto.FlatMapGroupsWithState): LogicalPlan = {
+    val baseRel = transformRelation(rel.getInput)
+    val udf = rel.getFunc.getScalarScalaUdf
+    val udfPacket =
+      Utils.deserialize[UdfPacket](
+        udf.getPayload.toByteArray,
+        SparkConnectArtifactManager.classLoaderWithArtifacts)
+    assert(udfPacket.inputEncoders.size == 1)
+    val iEnc = ExpressionEncoder(udfPacket.inputEncoders.head)
+    val rEnc = ExpressionEncoder(udfPacket.outputEncoder)
+
+    val deserializer = UnresolvedDeserializer(iEnc.deserializer)
+    val deserialized = DeserializeToObject(deserializer, generateObjAttr(iEnc), baseRel)
+
+    val logicalPlan = deserialized.child
+    val dataAttributes = logicalPlan.output
+
+    val groupingAttributes = AppendColumns(udfPacket.function.asInstanceOf[Any => Any], logicalPlan)
```

Comment:
Logics for getting `dataAttributes` and `groupingAttributes`. The issue for retrieving them inside `transformFlatMapGroupsWithState` is that users may want to set initial state which might not be able to handled here since it requires the `dataAttributes` and `groupingAttributes` from the initial state as well. ([code reference](https://github.com/apache/spark/blob/6382a3be8f6a937412c4e23a92ab440f7ba80bdf/sql/core/src/main/scala/org/apache/spark/sql/KeyValueGroupedDataset.scala#L400-L402))